os/20211014-s390x-secure-execution.md

```diff
@@ -0,0 +1,157 @@
+# IBM Secure Execution for Linux
+---
+
+# Theory of operation
+
+[IBM Secure Execution for Linux®](https://www.ibm.com/docs/en/linux-on-systems?topic=virtualization-introducing-secure-execution-linux) is a hardware-based security technology that is
+built into the IBM z15™ and LinuxONE III generation systems. It is designed to
+provide scalable isolation for individual workloads to help protect them from not
+only external attacks, but also insider threats. In traditional x86 ring architectures,
+the host can access the memory and data of guest applications freely, leading to the
+potential for malicious software to be proliferated throughout the entire system.
+Isolation between host and guest environments is necessary to help prevent system compromise.
+Users can enable Secure Execution feature on IBM zKVM Host and Guest VMs to isolate them from each other.
+
+That means:
+1. An encrypted Linux image, called `sd-boot`, (kernel + initramfs + cmdline) is created using the host public key and a customer-specific key. Since the encryption keys are stored on the IBM Z hardware and firmware, the encrypted image can only be executed in a virtual machine on the host(s) it has been prepared for, and the image can’t be decrypted or tampered with outside of the designated host(s).
```

Comment:
`host public key` (i.e. universal key) is used during `cosa build`, and indeed it's not public at all . than during `firstboot` we regenerate `sd-boot` with customer's key(s)

Reply:
I'm confused about the "customer keys" part.  My understanding is that we generate a random key for disk encryption, and we use the host keys of the target machines for sd-boot encryption, but that the customer does not supply their own key for the sd-boot image.

Reply:
sorry, i wrote not  clear. so, we have 2 sets of keys - 1 for SE, and 1 for LUKS2. As you said, LUKS keys are random, and customer/user doesn't touch them. But for SE customer has its own key(s), which are listed in `ignition_config.ign` , and these keys are replacing universal hostkey from IBM

Reply:
To be clear, those keys are the public keys for the hosts the customer intends to run on, correct?  "Customer-specific key" feels confusing in that context, since the customer doesn't control the corresponding private keys.

Reply:
yep, fixed